src/Microsoft.TestPlatform.Build/Tasks/TestTaskUtils.cs

```diff
@@ -0,0 +1,222 @@
+
+            string vsTestVerbosity = "minimal";
+            if (normalTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "normal";
+            }
+            else if (quietTestLogging.Contains(task.VSTestVerbosity.ToLowerInvariant()))
+            {
+                vsTestVerbosity = "quiet";
+            }
+
+            builder.AppendSwitchUnquotedIfNotNull("--logger:", $"Console;Verbosity={vsTestVerbosity}");
+        }
+
+        if (task.VSTestBlame || task.VSTestBlameCrash || task.VSTestBlameHang)
+        {
+            var dumpArgs = new List<string>();
+            if (task.VSTestBlameCrash || task.VSTestBlameHang)
+            {
+                if (task.VSTestBlameCrash)
+                {
+                    dumpArgs.Add("CollectDump");
+                    if (task.VSTestBlameCrashCollectAlways)
+                    {
+                        dumpArgs.Add($"CollectAlways={task.VSTestBlameCrashCollectAlways}");
+                    }
+
+                    if (!task.VSTestBlameCrashDumpType.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"DumpType={task.VSTestBlameCrashDumpType}");
+                    }
+                }
+
+                if (task.VSTestBlameHang)
+                {
+                    dumpArgs.Add("CollectHangDump");
+
+                    if (!task.VSTestBlameHangDumpType.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"HangDumpType={task.VSTestBlameHangDumpType}");
+                    }
+
+                    if (!task.VSTestBlameHangTimeout.IsNullOrEmpty())
+                    {
+                        dumpArgs.Add($"TestTimeout={task.VSTestBlameHangTimeout}");
+                    }
+                }
+            }
+
+            if (dumpArgs.Any())
+            {
+                builder.AppendSwitchIfNotNull("--Blame:", string.Join(";", dumpArgs));
+            }
+            else
+            {
+                builder.AppendSwitch("--Blame");
+            }
+        }
+
+        if (task.VSTestCollect != null && task.VSTestCollect.Any())
+        {
+            foreach (var arg in task.VSTestCollect)
+            {
+                if (arg != null)
+                {
+                    // For collecting code coverage, argument value can be either "Code Coverage" or "Code Coverage;a=b;c=d".
+                    // Split the argument with ';' and compare first token value.
+                    var tokens = arg.Split(';');
+
+                    if (arg.Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase) ||
+                        tokens[0].Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase))
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
+                }
+                builder.AppendSwitchIfNotNull("--collect:", arg);
```

Comment:
It seems it could be simplified by inverting the condition.

```suggestion
                if (arg == null)
                {  
                    continue;
                }

                // For collecting code coverage, argument value can be either "Code Coverage" or "Code Coverage;a=b;c=d".
                // Split the argument with ';' and compare first token value.
                var tokens = arg.Split(';');

                if (arg.Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase) ||
                    tokens[0].Equals(codeCoverageString, StringComparison.OrdinalIgnoreCase))
                {
                    isCollectCodeCoverageEnabled = true;
                }

                builder.AppendSwitchIfNotNull("--collect:", arg);
```